src/main/java/com/hazelcast/kubernetes/KubernetesApiEndpointResolver.java

```diff
@@ -56,49 +53,59 @@
+        return getSimpleDiscoveryNodes(client.endpoints());
     }
 
-    private List<DiscoveryNode> getSimpleDiscoveryNodes(Endpoints endpoints) {
+    private List<DiscoveryNode> getSimpleDiscoveryNodes(List<Endpoint> endpoints) {
         List<DiscoveryNode> discoveredNodes = new ArrayList<DiscoveryNode>();
-        resolveNotReadyAddresses(discoveredNodes, endpoints.getNotReadyAddresses());
-        resolveAddresses(discoveredNodes, endpoints.getAddresses());
+        resolveAddresses(discoveredNodes, endpoints);
         return discoveredNodes;
     }
 
-    private void resolveNotReadyAddresses(List<DiscoveryNode> discoveredNodes, List<EntrypointAddress> notReadyAddresses) {
+    private void resolveNotReadyAddresses(List<DiscoveryNode> discoveredNodes, List<Endpoint> notReadyAddresses) {
         if (Boolean.TRUE.equals(resolveNotReadyAddresses)) {
             resolveAddresses(discoveredNodes, notReadyAddresses);
         }
     }
 
-    private void resolveAddresses(List<DiscoveryNode> discoveredNodes, List<EntrypointAddress> addresses) {
-        for (EntrypointAddress address : addresses) {
+    private void resolveAddresses(List<DiscoveryNode> discoveredNodes, List<Endpoint> addresses) {
+        for (Endpoint address : addresses) {
             addAddress(discoveredNodes, address);
         }
     }
 
-    private void addAddress(List<DiscoveryNode> discoveredNodes, EntrypointAddress entrypointAddress) {
-        String ip = entrypointAddress.getIp();
-        InetAddress inetAddress = mapAddress(ip);
-        int port = port(entrypointAddress);
-        Address address = new Address(inetAddress, port);
-        discoveredNodes.add(new SimpleDiscoveryNode(address, entrypointAddress.getAdditionalProperties()));
-        if (logger.isFinestEnabled()) {
-            logger.finest("Found node service with address: " + address);
+    private void addAddress(List<DiscoveryNode> discoveredNodes, Endpoint endpoint) {
+        if (Boolean.TRUE.equals(resolveNotReadyAddresses) || endpoint.isReady()) {
+            Address address = createAddress(endpoint.getPrivateAddress());
+            Address publicAddress = createAddress(endpoint.getPublicAddress());
+            discoveredNodes
+                    .add(new SimpleDiscoveryNode(address, publicAddress, endpoint.getAdditionalProperties()));
+            if (logger.isFinestEnabled()) {
+                logger.finest("Found node service with address: " + address);
+            }
         }
     }
 
-    private int port(EntrypointAddress entrypointAddress) {
+    private Address createAddress(KubernetesClient.EndpointAddress address) {
+        if (address == null) {
+            return null;
+        }
+        String ip = address.getIp();
+        InetAddress inetAddress = mapAddress(ip);
+        int port = port(address);
+        return new Address(inetAddress, port);
+    }
+
+    private int port(KubernetesClient.EndpointAddress address) {
         if (this.port > 0) {
```

Comment:
I see that `service-port` field is a configuration option which is not available in README. Do you know what is the use case for it. I remember we discussed that but I don't remember the details anymore 😄

Reply:
The only valid use case I see is if you have a service which points to two ports, e.g. you use hazelcast embedded into your Spring Boot application and one service which points to both `5701` and `8080`. Then, if you decide not to use `5701` port for some reason, but `5702`, then you could use this property.

Saying that, I think it's super edge use case and I don't believe anyone uses it ;) Not sure it's even worth documenting it...

Reply:
I actually wanted to remove it. If not documented then I don't think anyone using it. It just makes the code complicated.  We can create a deprecation issue and remove it in the next release. Then it will be documented. We can always put it back if someone asks.

Reply:
Ok, I've created an issue to remove all the detracted properties #129. Apart from `service-port`, we should also remove labeling services as described here: #115